bundles/org.openhab.binding.haassohnpelletoven/src/main/java/org/openhab/binding/haassohnpelletoven/internal/HaassohnpelletstoveHandler.java

```diff
@@ -0,0 +1,390 @@
+
+    private void updateLinkedChannels() {
+
+        verifyLinkedChannel(CHANNEL_isTemp);
+        verifyLinkedChannel(CHANNEL_mode);
+        verifyLinkedChannel(CHANNEL_prg);
+        verifyLinkedChannel(CHANNEL_spTemp);
+        verifyLinkedChannel(CHANNEL_ecoMode);
+        verifyLinkedChannel(CHANNEL_ignitions);
+        verifyLinkedChannel(CHANNEL_maintenanceIn);
+        verifyLinkedChannel(CHANNEL_cleaningIn);
+        verifyLinkedChannel(CHANNEL_consumption);
+        verifyLinkedChannel(CHANNEL_onTime);
+
+        if (!linkedChannels.isEmpty()) {
+            logger.debug("Start automatic refreshing");
+
+            updateOvenData(null);
+
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId());
+            }
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+    }
+
+    private void verifyLinkedChannel(String channelID) {
+        if (isLinked(channelID) && !linkedChannels.containsKey(channelID)) {
+            linkedChannels.put(channelID, true);
+        }
+    }
+
+    @Override
+    public void dispose() {
+
+        logger.debug("Disposing Haas and Sohn Pellet stove handler.");
+        stopScheduler();
+    }
+
+    private void stopScheduler() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            job.cancel(true);
+        }
+        refreshJob = null;
+    }
+
+    /**
+     * Start the job refreshing the oven status
+     */
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job == null || job.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+
+                        updateOvenData(null);
+
+                        for (Channel channel : getThing().getChannels()) {
+                            updateChannel(channel.getUID().getId());
+                        }
+                    } catch (Exception e) {
+                        logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            @SuppressWarnings("null")
+            int period = config.refreshRate;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, period, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+
+        if (!automaticRefreshing) {
+            logger.debug("Start automatic refreshing");
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+
+        verifyLinkedChannel(channelUID.getId());
+        updateChannel(channelUID.getId());
+    }
+
+    @Override
+    public void channelUnlinked(ChannelUID channelUID) {
+
+        if (linkedChannels.containsKey(channelUID.getId())) {
+
+            linkedChannels.remove(channelUID.getId());
+        }
```

Comment:
No harm in removing a key that doesn't exist.
```suggestion
            linkedChannels.remove(channelUID.getId());
```